packages/relic/lib/src/headers/special/accept_encoding_header.dart

```diff
@@ -0,0 +1,55 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Accept-Encoding header.
+///
+/// This header specifies the content encoding that the client can understand.
+class AcceptEncodingHeader {
+  /// The list of encodings that are accepted.
+  final List<EncodingQuality> encodings;
+
+  /// Constructs an instance of [AcceptEncodingHeader] with the given encodings.
+  AcceptEncodingHeader(this.encodings);
+
+  /// Parses the Accept-Encoding header value and returns an [AcceptEncodingHeader] instance.
+  factory AcceptEncodingHeader.parse(List<String> values) {
+    var parts = values.splitTrimAndFilterUnique();
+
+    if (parts.isEmpty) {
+      throw FormatException(
+        'Invalid Accept-Encoding header: value cannot be empty',
+      );
+    }
+
+    var encodings = parts.map((e) {
+      var parts = e.split(';q=');
+      var encoding = parts[0].trim().toLowerCase();
+      var quality = switch (parts.length > 1) {
+        true => double.parse(parts[1].trim()),
+        false => 1.0,
+      };
+      return EncodingQuality(encoding, quality);
+    }).toList();
+
+    return AcceptEncodingHeader(encodings);
+  }
+
+  /// Converts the [AcceptEncodingHeader] instance into a string representation suitable for HTTP headers.
+  @override
+  String toString() => encodings.map((e) => e.toString()).join(', ');
```

Comment:
Question: Should we have a prefix here that explains these encodings are part of the `AcceptEncodingHeader`? Just thinking if we print multiple headers at the same time that contain encodings it will not be clear that these specific encodings are part of the `AcceptEncodingHeader`.

Reply:
something like this:

```
 /// Converts the [AcceptEncodingHeader] instance into a string representation suitable for HTTP headers.
  String toHeaderString() => encodings.map((e) => e.toHeaderString()).join(', ');

  @override
  String toString() => 'AcceptEncodingHeader(encodings: $encodings)';
```

Reply:
Yeah, that makes sense!